File: ios/SwiftKotlination/SwiftKotlinationTests/Sources/Model/Mocks/ImageManagerMock.swift
import XCTest
@testable import SwiftKotlination

struct ImageManagerMock {
    var result: Result<UIImage, Error>

    init(result: Result<UIImage, Error> = .failure(NetworkError.invalidResponse)) {
        self.result = result
    }
}

extension ImageManagerMock: ImageManagerProtocol {
    func image(with url: String, _ observer: @escaping Observer<UIImage>) -> Disposable {
        observer(result)
        return Disposable {}
    }
}

File: android/SwiftKotlination/app/src/test/kotlin/fr/jhandguy/swiftkotlination/model/mocks/ImageManagerMock.kt
package fr.jhandguy.swiftkotlination.model.mocks

import android.graphics.Bitmap
import fr.jhandguy.swiftkotlination.model.ImageManagerInterface
import fr.jhandguy.swiftkotlination.observer.Disposable
import fr.jhandguy.swiftkotlination.observer.Observer
import fr.jhandguy.swiftkotlination.observer.Result

class ImageManagerMock(val result: Result<Bitmap>) : ImageManagerInterface {
    override suspend fun image(url: String, observer: Observer<Bitmap>): Disposable {
        observer(result)

        return Disposable {}
    }
}
